test/python/circuit/library/test_pauli_feature_map.py

```diff
@@ -295,5 +304,262 @@ def test_entanglement_not_specified(self):
+            circuit.cx(qubit1, qubit2)
+
+        #      ┌───┐┌────────┐                                         ┌───┐┌────────┐»
+        # q_0: ┤ H ├┤ P(0.4) ├──■─────────────────■────■────────────■──┤ H ├┤ P(0.4) ├»
+        #      ├───┤└┬──────┬┘┌─┴─┐┌───────────┐┌─┴─┐  │            │  └───┘└────────┘»
+        # q_1: ┤ H ├─┤ P(2) ├─┤ X ├┤ P(12.599) ├┤ X ├──┼────────────┼────■────────────»
+        #      ├───┤┌┴──────┤ └───┘└───────────┘└───┘┌─┴─┐┌──────┐┌─┴─┐┌─┴─┐ ┌──────┐ »
+        # q_2: ┤ H ├┤ P(2π) ├────────────────────────┤ X ├┤ P(0) ├┤ X ├┤ X ├─┤ P(0) ├─»
+        #      └───┘└───────┘                        └───┘└──────┘└───┘└───┘ └──────┘ »
+        # «                                                                              »
+        # «q_0: ─────────────────────■─────────────────■────■────────────■───────────────»
+        # «          ┌───┐ ┌──────┐┌─┴─┐┌───────────┐┌─┴─┐  │            │               »
+        # «q_1: ──■──┤ H ├─┤ P(2) ├┤ X ├┤ P(12.599) ├┤ X ├──┼────────────┼────■──────────»
+        # «     ┌─┴─┐├───┤┌┴──────┤└───┘└───────────┘└───┘┌─┴─┐┌──────┐┌─┴─┐┌─┴─┐┌──────┐»
+        # «q_2: ┤ X ├┤ H ├┤ P(2π) ├───────────────────────┤ X ├┤ P(0) ├┤ X ├┤ X ├┤ P(0) ├»
+        # «     └───┘└───┘└───────┘                       └───┘└──────┘└───┘└───┘└──────┘»
+        # «
+        # «q_0: ─────
+        # «
+        # «q_1: ──■──
+        # «     ┌─┴─┐
+        # «q_2: ┤ X ├
+        # «     └───┘
+        ref = QuantumCircuit(3)
+        for _ in range(2):
+            ref.h([0, 1, 2])
+            for i in range(3):
+                ref.p(2 * times[i], i)
+            zz_evolution(ref, 0, 1)
+            zz_evolution(ref, 0, 2)
+            zz_evolution(ref, 1, 2)
+
+        self.assertTrue(Operator(encoding).equiv(ref))
+
+    @combine(entanglement=["linear", "reverse_linear", "pairwise"])
+    def test_zz_entanglement(self, entanglement):
+        """Test the ZZ feature map works with pairwise, linear and reverse_linear entanglement."""
+        num_qubits = 5
+        encoding = zz_feature_map(num_qubits, entanglement=entanglement, reps=1)
+        ops = encoding.count_ops()
+        expected_ops = {"h": num_qubits, "p": 2 * num_qubits - 1, "cx": 2 * (num_qubits - 1)}
+        self.assertEqual(ops, expected_ops)
+
+    def test_pauli_alpha(self):
+        """Test  Pauli rotation factor (getter, setter)."""
+        alpha = 1.234
+
+        # this is needed as the outcoming Rust circuit has no qreg
+        encoding = QuantumCircuit(1)
+        encoding.compose(pauli_feature_map(1, alpha=alpha, paulis=["z"], reps=1), inplace=True)
+
+        ref = QuantumCircuit(1)
+        ref.h(0)
+        ref.p(alpha * encoding.parameters[0], 0)
+
+        self.assertEqual(ref, encoding)
+
+    def test_zzfeaturemap_raises_if_too_small(self):
+        """Test the ``ZZFeatureMap`` raises an error if the number of qubits is smaller than 2."""
+        with self.assertRaises(QiskitError):
+            _ = zz_feature_map(1)
```

Comment:
Ah that was because I didn't know how to raise `ValueError`s at the time, I can try changing it (and see whether the Rust compiler is happy with returning `PyValueError`s and `QiskitError`s from the same function)

Reply:
I left them as `QiskitError` as that's coming from the entanglement functions -- we can change it there if we want to in a separate PR, but I thought it's best to not touch it in the Pauli feature map PR 🙂